rs/nns/cmc/src/main.rs

```diff
@@ -3158,4 +3208,138 @@ mod tests {
         )
         .expect("The CMC canister interface is not compatible with the cmc.did file");
     }
+
+    #[test]
+    fn test_transaction_has_expected_memo_happy() {
+        // Not relevant to this test.
+        let operation = Operation::Mint {
+            to: AccountIdentifier::new(PrincipalId::new_user_test_id(668_857_347), None),
+            amount: Tokens::from_e8s(123_456),
+        };
+
+        // Case A: Legacy memo is used.
+        let transaction_with_legacy_memo = Transaction {
+            memo: Memo(42),
+            icrc1_memo: None,
+
+            // Irrelevant to this test.
+            operation: operation.clone(),
+            created_at_time: None,
+        };
+
+        assert_eq!(
+            transaction_has_expected_memo(&transaction_with_legacy_memo, Memo(42),),
+            Ok(()),
+        );
+
+        // Case B: icrc1_memo is used.
```

Comment:
```suggestion
        // Case B: when both are specified, icrc1_memo is used.
```

Reply:
What I meant here is that the user used icrc1 to make/perform the transfer/transaction. Reworded.

iiuc, it is not possible for a user to specify both legacy memo and icrc1_memo. Instead, they either use the legacy API or icrc1. If they use icrc1, then when you fetch the transaction, the legacy memo field will always implicitly be 0.